cmd/bootkube/render.go

```diff
@@ -65,6 +71,12 @@ func init() {
 	cmdRender.Flags().StringVar(&renderOpts.etcdCAPath, "etcd-ca-path", "", "Path to an existing PEM encoded CA that will be used for TLS-enabled communication between the apiserver and etcd. Must be used in conjunction with --etcd-certificate-path and --etcd-private-key-path, and must have etcd configured to use TLS with matching secrets.")
 	cmdRender.Flags().StringVar(&renderOpts.etcdCertificatePath, "etcd-certificate-path", "", "Path to an existing certificate that will be used for TLS-enabled communication between the apiserver and etcd. Must be used in conjunction with --etcd-ca-path and --etcd-private-key-path, and must have etcd configured to use TLS with matching secrets.")
 	cmdRender.Flags().StringVar(&renderOpts.etcdPrivateKeyPath, "etcd-private-key-path", "", "Path to an existing private key that will be used for TLS-enabled communication between the apiserver and etcd. Must be used in conjunction with --etcd-ca-path and --etcd-certificate-path, and must have etcd configured to use TLS with matching secrets.")
+	cmdRender.Flags().StringVar(&renderOpts.selfHostedEtcdClientCAPath, "self-hosted-etcd-client-ca", "", "Path to an existing PEM encoded CA that will be used for self-hosted etcd's client port.")
+	cmdRender.Flags().StringVar(&renderOpts.selfHostedEtcdClientCertPath, "self-hosted-etcd-client-cert", "", "Path to an existing certificate that will be used for self-hosted etcd's client port.")
+	cmdRender.Flags().StringVar(&renderOpts.selfHostedEtcdClientKeyPath, "self-hosted-etcd-client-key", "", "Path to an existing private key that will be used for self-hosted etcd's client port.")
+	cmdRender.Flags().StringVar(&renderOpts.selfHostedEtcdPeerCAPath, "self-hosted-etcd-peer-ca", "", "Path to an existing PEM encoded CA that will be used for  self-hosted etcd's peer port.")
+	cmdRender.Flags().StringVar(&renderOpts.selfHostedEtcdPeerCertPath, "self-hosted-etcd-peer-cert", "", "Path to an existing certificate that will be used for self-hosted etcd's peer port.")
```

Comment:
> Will render generate certs in any scenario, or do you always have to specify all 6 flags?

No. The CA could not be used to generate certs. We have this discussion previously in etcd operator on TLS story like taking only the CA and generating certs on our own. But it doesn't work. We couldn't make assumptions on this. But I'm open to any suggestions to make it more automated.